src/main/java/io/nats/client/impl/DispatcherFactory.java

```diff
@@ -0,0 +1,28 @@
+// Copyright 2023 The NATS Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package io.nats.client.impl;
+
+import io.nats.client.MessageHandler;
+
+/**
+ * !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+ * WARNING: THIS CLASS IS PUBLIC BUT ITS API IS NOT GUARANTEED
+ * TO BE BACKWARD COMPATIBLE AS IT IS PURELY AN INTERNAL CLASS
+ * !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+ */
+public class DispatcherFactory {
+    NatsDispatcher createDispatcher(NatsConnection conn, MessageHandler handler) {
+        return new NatsDispatcher(conn, handler);
+    }
+}
```

Comment:
This was made a class, not an interface. I'm trying to limit exposure of public things, note the warning, and a public interface required making NatsDispatcher and NatsConnection public